rootfs/opt/kubectl/example/GoogleBackup/yaml-builder.py

```diff
@@ -0,0 +1,150 @@
+    # add formatter to ch
+    console.setFormatter(formatter)
+
+    _LOG.handlers = [console]
+    _LOG.setLevel(levels.get(verbosity, logging.DEBUG))
+
+if __name__ == "__main__":
+    try:
+
+        parser = argparse.ArgumentParser(
+            description='Create k8s pods configuration file from template')
+
+        parser.add_argument('-v', '--verbose', dest='verbose', default=[],
+                            action='append_const', const=None,
+                            help='More verbose output, can use several times.')
+        parser.add_argument('-T', '--tier', dest='tier',
+                            required=True,
+                            help='tier to backup (qserv, czar, repl-db)')
+        parser.add_argument('-r', '--replica-count', dest='replica_count',
+                            metavar='N', type=int,
+                            required=False, default=1,
+                            help='number of replica to backup, default to %(default)s')
+        parser.add_argument('-R', '--restore', dest='restore', action='store_true',
+                             default=False,
+                             help='Launch restore procedure instead of backup by default'
+                             )
+        parser.add_argument('-t', '--template', dest='templateFile',
+                            required=True, metavar='PATH',
+                            help='yaml template file')
+        parser.add_argument('-o', '--output', dest='yamlFile',
+                            required=True, metavar='PATH',
+                            help='pod configuration file, in yaml')
+        parser.add_argument('-V', '--volumeclaimname', dest='volume_claim_name',
+                            required=False, default="qserv-data",
+                            help='name for volume claim')
+
+        args = parser.parse_args()
+
+        _config_logger(args.verbose)
+
+        config = configparser.RawConfigParser()
+
+        with open(args.templateFile, 'r') as f:
+            yaml_data = yaml.load(f)
+
+        yaml.add_representer(str, _str_presenter)
+
+        yaml_data_tpl = yaml_data['spec']['template']['spec']
+
+        tier = args.tier
+
+        yaml_data['metadata']['name'] = tier
+        yaml_data['metadata']['labels']['tier'] = tier
+        yaml_data['spec']['replicas'] = args.replica_count
+        yaml_data['spec']['selector']['matchLabels']['tier'] = tier
+        yaml_data['spec']['template']['metadata']['labels']['tier'] = tier
+
+        if args.restore:
+            cmd = '/config-backup/restore.sh'
+        else:
+            cmd = '/config-backup/backup.sh'
+        yaml_data['spec']['template']['spec']['containers'][0]['command'] = [ 'sh', cmd ]
+        yaml_data['spec']['template']['spec']['containers'][0]['volumeMounts'][0]['name'] = args.volume_claim_name
+        yaml_data['spec']['volumeClaimTemplates'][0]['metadata']['name'] = args.volume_claim_name
```

Comment:
Instead of digging deep in here and using the YAML and trying to find where things are, why not just use jinja and replace things?  It looks like you're almost there with the placeholder values you have like <TIER> etc.  I thought this script was going to do a string replace for the things in the <>, but it looks like even if you put a value in there, this will overwrite it?

If you do a jinja rendering, you can just do this as one big line and pass in a dict of things to replace (and then you just have to edit the template, not this part as well)

Reply:
Jinja will also handle writing out the files, and going over an entire directory of templates very easily.

Reply:
Jinja might be a good solution, but I may also use helm and k8s CRD instead... Both Robinson@Google and my student found the yaml stuff weird at first sight. But both of them found it was a good solution for prototyping on fast moving Qserv architecture. Furthermore Robinson told me it was great not to add external dependency and my student finally found this yaml lib very powerful... So I may stick with this for prototyping and move to something better (helm and CRD looks ok) when k8s/Qserv integration  will be stable.